subsys/logging/log_core.c

```diff
@@ -530,7 +532,7 @@ void log_panic(void)
 
 	if (!IS_ENABLED(CONFIG_LOG_IMMEDIATE)) {
 		/* Flush */
-		while (log_process(false) == true) {
+		while (log_process(false)) {
```

Comment:
This change is not reflected in the commit message; why was this required?

Reply:
In general, comparing `variable == true` is not good practice. Identifier `true` is one specific value. C assumes that any non-zero value is true.

Reply:
yes, that was suggested by @MarekPieta

Reply:
@MarekPieta I was, interested in knowing why this change was included in this patch, without being related to the actual bug-fix, and without being reflected in the commit message :)

But since you bring this up, here's the API declaration:

```
/**
 * @brief Process one pending log message.
 *
 * @param bypass If true message is released without being processed.
 *
 * @retval true There is more messages pending to be processed.
 * @retval false No messages pending.
 */
bool log_process(bool bypass);
```
The function is supposed to return false, or true, so checking explicitly for true, here, does not seem to be a problem IMHO.

Reply:
@ioannisg i've removed that change. PR contains only changes related to commit message.